File: boost_python/wtest.c
#include <boost/python/def.hpp>
#include <boost/python/class.hpp>
#include <boost/python/module.hpp>

#include "wtest.h"

BOOST_PYTHON_MODULE(_wtest)
{
    using namespace boost::python;

    def("empty_func", empty_func);
    def("simple_func", simple_func);
    def("many_params", many_params);
    def("struct_param", struct_param);

    class_<Data>("Data")
        .def_readwrite("field1", &Data::field1)
        .def_readwrite("field3", &Data::field2)
        .def_readwrite("field3", &Data::field3)
        .def_readwrite("field4", &Data::field4)
        .def_readwrite("field5", &Data::field5)
        .def_readwrite("field6", &Data::field6)
        .def_readwrite("field7", &Data::field7);
}

File: cython/wtest.py
import _wtest

class Module(object):
    pass

class Data(object):
    def __init__(self, field1, field2, field3, field4, field5, field6, field7):
        self.field1 = field1
        self.field2 = field2
        self.field3 = field3
        self.field4 = field4
        self.field5 = field5
        self.field6 = field6
        self.field7 = field7

def get_module():
    mod = Module()
    mod.empty_func = _wtest.py_empty_func
    mod.simple_func = _wtest.py_simple_func
    mod.many_params = _wtest.py_many_params
    mod.struct_param = _wtest.py_struct_param
    mod.Data = Data
    return mod
